suite-native/atoms/src/Select/Select.tsx

```diff
@@ -23,36 +28,68 @@ export const Select = <TItemValue extends SelectItemValue>({
     selectLabel,
     selectValue,
     onSelectItem,
+    isConfirmable = false,
     testID,
 }: SelectProps<TItemValue>) => {
     const { bottomSheetRef, openModal, closeModal } = useBottomSheetModal();
 
-    const selectedItem = useMemo(
-        () => items.find(item => item.value === selectValue),
-        [selectValue, items],
+    const selectTriggerValue = useMemo(
+        () => items.find(item => item.value === selectValue)?.label ?? null,
+        [items, selectValue],
     );
-    const handleSelectItem = (itemValue: TItemValue) => {
+
+    const [selectedItemValue, setSelectedItemValue] = useState(selectValue);
+    const [isConfirmButtonVisible, setIsConfirmButtonVisible] = useState(false);
+
+    const openBottomSheet = () => {
+        setSelectedItemValue(selectValue);
+        setIsConfirmButtonVisible(false);
+        openModal();
+    };
+
+    const confirmSelection = (itemValue: TItemValue) => {
         onSelectItem(itemValue);
         closeModal();
     };
 
+    const handleSelectItem = (itemValue: TItemValue) => {
+        setSelectedItemValue(itemValue);
+        if (isConfirmable) {
+            setIsConfirmButtonVisible(itemValue !== selectValue);
```

Comment:
other nit: 

The setter has no effect for the `else` branch

```suggestion
        if (isConfirmable) {
            setSelectedItem(item);
            setIsConfirmButtonVisible(item.value !== selectValue);
        }
```

Reply:
Right, fixed.